### ISSUE TITLE
Commit sctruct on events

### ISSUE THREAD

**Issue by leandro-lugaresi** [xanzy/go-gitlab#139]
The commit data from events is a little different from other commit data from API
This is the commit data from PushEvent:
```json
    {
      "id": "da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
      "message": "fixed readme",
      "timestamp": "2012-01-03T23:36:29+02:00",
      "url": "http://example.com/mike/diaspora/commit/da1560886d4f094c3e6c9ef40349f7d38b5d27d7",
      "author": {
        "name": "GitLab dev user",
        "email": "gitlabdev@dv6700.(none)"
      },
      "added": ["CHANGELOG"],
      "modified": ["app/controller/application.rb"],
      "removed": []
    }
```
And this json is used in other events like:
https://github.com/xanzy/go-gitlab/blob/master/events_test.go#L151
https://github.com/xanzy/go-gitlab/blob/master/events_test.go#L240
Can we use this data in the Commit struct or create another struct?

**Comment by svanharmelen** [comment]
Hi @leandro-lugaresi I'm afraid I don't understand your question. Could you rephrase and/or explain a bit more what you want to know? Thanks!

**Comment by leandro-lugaresi** [comment]
Sorry.
The commit data from webhooks have some diffs from Commit struct used by gitlab API.
The struct we are using is [https://github.com/xanzy/go-gitlab/blob/master/commits.go#L36](https://github.com/xanzy/go-gitlab/blob/master/commits.go#L36) but the json is different for commits from webhooks.

**Comment by svanharmelen** [comment]
@leandro-lugaresi I understand now. Don't know what the best solution is... Adding the data directly into the structs that need it is probably the most clean. It will prevent having 2 structs that are almost the same with probably also almost the same name. But it then does mean the code (the struct fields) are then duplicated a couple of times...

Still think it is the best solution to just add the fields directly into the event structs that need it...

**Comment by svanharmelen** [comment]
Closing this issue as the question seems to be answered and no conversation is happening anymore on this topic. Please reopen if you have additional questions/issues.
